dependencies/hpdf/CMakeLists.txt

```diff
@@ -0,0 +1,26 @@
+
+
+# Sources
+file(GLOB SRCS_G "src/*.c")
+POCO_SOURCES(SRCS hpdf ${SRCS_G})
+
+# Headers
+file(GLOB_RECURSE HDRS_G "src/*.h")
+POCO_HEADERS(SRCS hpdf ${HDRS_G})
+
+add_library(_BUNDLED_HPDF OBJECT EXCLUDE_FROM_ALL ${SRCS})
+
+set_property(TARGET _BUNDLED_HPDF PROPERTY POSITION_INDEPENDENT_CODE ON)
+
+target_include_directories(_BUNDLED_HPDF
+	PUBLIC
+		$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>
+)
+
+if (POCO_UNBUNDLED)
+	target_link_libraries(_BUNDLED_HPDF PRIVATE PNG::PNG)
+else()
+	target_link_libraries(_BUNDLED_HPDF PRIVATE "$<BUILD_LOCAL_INTERFACE:PNG::PNG>")
```

Comment:
Is there an alternate solution to the BUILD_LOCAL_INTERFACE generator expression here? I think it would be good if Poco kept the cmake minimum to 3.15, or at least not jumping all the way to 3.26.

Reply:
I was not able to find other solution that does not involve writing own custom CMake scripts. And CMake 3.26 is the first version that has clean solution to link private Poco library and not get undefined export errors.